logger/tests/helpers.rs

```diff
@@ -0,0 +1,116 @@
+use portpicker::pick_unused_port;
+use std::{
+    process::Command,
+    thread::sleep,
+    time::{Duration, SystemTime},
+};
+
+const PG_CONTAINER_NAME: &str = "shuttle_logger_test_pg";
+
+pub struct DockerInstance {
+    pub container_name: &'static str,
+    pub uri: String,
+}
+
+struct Config<'a> {
+    container_name: &'a str,
+    image: &'a str,
+    engine: &'a str,
+    port: &'a str,
+    env: Vec<&'a str>,
+    is_ready_cmd: Vec<&'a str>,
+}
+
+impl DockerInstance {
+    pub fn new() -> Self {
+        let Config {
+            engine,
+            env,
+            image,
+            is_ready_cmd,
+            port,
+            container_name,
+        } = Config {
+            container_name: PG_CONTAINER_NAME,
+            // The postgres version should always be in sync with the prod RDS version.
+            image: "docker.io/library/postgres:14",
+            engine: "postgres",
+            port: "5432",
+            env: vec!["POSTGRES_PASSWORD=password", "PGUSER=postgres"],
+            is_ready_cmd: vec!["exec", PG_CONTAINER_NAME, "pg_isready"],
+        };
+
+        let host_port = pick_unused_port().unwrap();
+        let port_binding = format!("{}:{}", host_port, port);
+
+        let mut args = vec!["run", "--rm", "--name", container_name, "-p", &port_binding];
+
+        args.extend(env.iter().flat_map(|e| ["-e", e]));
+
+        args.push(image);
+
+        Command::new("docker").args(args).spawn().unwrap();
+
+        Self::wait_ready(Duration::from_secs(120), &is_ready_cmd);
+
+        // TODO: find out why this is needed for tests to pass reliably.
+        sleep(Duration::from_millis(300));
```

Comment:
The postgres image enters the ready state and then reboots. So your first ready check would have passed. This sleep essentially gets it past this reboot.

You should be able to have a small delay after the first `wait_ready` and then have another `wait_ready` to have it work all the time.

Reply:
Oh, I see, it was like that in the provisioner, but the comment said it was because of MongoDB. I will revert it here then. Thanks!